models/user.py

```diff
@@ -81,15 +86,8 @@ def validate_reset_password(token):
         except ExpiredSignatureError:
             return None
 
-    @staticmethod
-    def _hash_pw(plaintext_password):
-        return bcrypt.hashpw(
-            bytes(plaintext_password, "utf-8"),
-            bcrypt.gensalt(current_app.config["WORK_FACTOR"]),
-        )
-
     def check_pw(self, plaintext_password):
-        return bcrypt.checkpw(bytes(plaintext_password, "utf-8"), self.hash_digest)
+        return bcrypt.checkpw(plaintext_password.encode("utf-8"), self.password)
```

Comment:
I think I would rather see this: And then delete the password getter. As password should not be needed outside of this class.
```suggestion
        return bcrypt.checkpw(plaintext_password.encode("utf-8"), self._password)
```